WooCommerce/Classes/Authentication/Prologue/LoginOnboardingPageViewController.swift

```diff
@@ -19,66 +17,38 @@ final class LoginOnboardingPageViewController: UIPageViewController {
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        dataSource = self
-        delegate = self
+        dataSource = pages.count > 1 ? self : nil
 
         if let firstPage = pages.first {
             setViewControllers([firstPage], direction: .forward, animated: false)
         }
+    }
 
-        configureUIBasedOnPageCount()
+    override func viewDidLayoutSubviews() {
+        if let pageControl = view.subviews.first(where: { $0 is UIPageControl }) as? UIPageControl {
+            configurePageControllerAppearance(pageControl: pageControl)
+        }
     }
 
     /// Shows the next page of content if it is not on the last page.
     /// - Returns: Whether it can go to the next page, if it has not reached the last page.
     func goToNextPageIfPossible() -> Bool {
-        let currentPage = pageControl.currentPage
+        let currentPage = dataSource?.presentationIndex?(for: self) ?? 0
         guard currentPage < pages.count - 1 else {
             return false
         }
-        pageControl.currentPage = currentPage + 1
-        setViewControllers([pages[pageControl.currentPage]], direction: .forward, animated: true)
+        setViewControllers([pages[currentPage + 1]], direction: .forward, animated: true)
         return true
     }
 }
 
 private extension LoginOnboardingPageViewController {
-    func configureUIBasedOnPageCount() {
-        if pages.count > 1 {
-            addPageControl()
-        } else {
-            // Sets data source to `nil` to disable scrolling.
-            dataSource = nil
-        }
-    }
-
     // MARK: Page Control Setup
     //
-    func addPageControl() {
-        pageControl.currentPageIndicatorTintColor = .gray(.shade5)
-        pageControl.pageIndicatorTintColor = .wooCommercePurple(.shade50)
+    func configurePageControllerAppearance(pageControl: UIPageControl) {
+        pageControl.currentPageIndicatorTintColor = .accent
```

Comment:
For the color itself, we could have just used `UIPageControler.appearance()`, but when trying to scale it by applying a `transform` to the appearance, it crashes, so instead we retrieve the page control from the subviews and keep the same logic as before.